psalm-baseline.xml

```diff
@@ -123,19 +123,24 @@
       <code><![CDATA[(object) $this->getArrayCopy()]]></code>
     </LessSpecificReturnStatement>
     <MixedAssignment>
-      <code>$this[$key]</code>
-      <code>$value</code>
+      <code><![CDATA[$this[$key]]]></code>
+      <code><![CDATA[$value]]></code>
     </MixedAssignment>
     <MoreSpecificReturnType>
-      <code>stdClass</code>
+      <code><![CDATA[stdClass]]></code>
     </MoreSpecificReturnType>
   </file>
+  <file src="src/Model/BSONIterator.php">
+    <MixedArgument>
+      <code><![CDATA[$this->options['typeMap']]]></code>
```

Comment:
Should we define array shape for the options?

Reply:
We had that. However, in the `param` declaration, it would have to be `array{typeMap?: array, ...}`, whereas for the property it's `array{typeMap: array, ...}`. With promoted properties, the same type is used for the parameter and property, leading to this error. Instead of adding custom logic to work around this, I decided to add the error to the baseline.